### PR TITLE
chore(primitives): fixed  `ExecutionAddress` validation

### PR DESCRIPTION
**Pull request by abi87**
fixed execution address validation and added UTs

## Summary by CodeRabbit

- **New Features**
	- Enhanced parsing logic for hex and JSON representations of `ExecutionAddress`, improving robustness and clarity.
	- Introduced a new error message for invalid hex string lengths.

- **Bug Fixes**
	- Improved error handling during unmarshalling of `ExecutionAddress`.

- **Tests**
	- Added comprehensive unit tests for marshalling and unmarshalling of `ExecutionAddress`, covering various edge cases.

### CODE REVIEW

File: mod/primitives/pkg/encoding/hex/bytes.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ import (
 	"github.com/berachain/beacon-kit/mod/errors"
 )
 
+var ErrInvalidHexStringLength = errors.New("invalid hex string length")
```
Comment (abi87) [inline mod/primitives/pkg/encoding/hex/bytes.go:29]: exported to assert exact errors in UTs
